<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Hw3</class>
 <widget class="QMainWindow" name="Hw3">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1062</width>
    <height>671</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Hw3</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <widget class="QSplitter" name="splitter">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <widget class="QWidget" name="widget_2" native="true">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
         <horstretch>2</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="styleSheet">
        <string notr="true">background-color: rgb(139, 139, 139);
border-color: rgb(8, 8, 8);</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="QLabel" name="lblBanner1">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="styleSheet">
           <string notr="true">background-color: rgb(85, 85, 127);
color: rgb(255, 255, 255);</string>
          </property>
          <property name="text">
           <string>Original Image</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="lblPic1">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="autoFillBackground">
           <bool>false</bool>
          </property>
          <property name="styleSheet">
           <string notr="true">background-color: rgb(215, 215, 215);
border-color: rgb(8, 8, 8);</string>
          </property>
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="btnOpen">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="styleSheet">
           <string notr="true">background-color: rgb(255, 255, 255);
color: rgb(0, 0, 0);
border-color: rgb(8, 8, 8);</string>
          </property>
          <property name="text">
           <string>Open file A</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="widget_3" native="true">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
         <horstretch>2</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="styleSheet">
        <string notr="true">background-color: rgb(21, 215, 215);
border-color: rgb(8, 8, 8);</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <widget class="QLabel" name="lblBanner2">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="styleSheet">
           <string notr="true">background-color: rgb(85, 85, 127);
color: rgb(255, 255, 255);</string>
          </property>
          <property name="text">
           <string>Filtered Image</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="lblPic2">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="autoFillBackground">
           <bool>false</bool>
          </property>
          <property name="styleSheet">
           <string notr="true">background-color: rgb(215, 215, 215);
border-color: rgb(8, 8, 8);</string>
          </property>
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeType">
           <enum>QSizePolicy::Fixed</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>30</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
      <widget class="QTabWidget" name="tabWidget">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="sizePolicy">
        <sizepolicy hsizetype="MinimumExpanding" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="baseSize">
        <size>
         <width>0</width>
         <height>0</height>
        </size>
       </property>
       <property name="currentIndex">
        <number>0</number>
       </property>
       <widget class="QWidget" name="tab">
        <attribute name="title">
         <string>Filtering</string>
        </attribute>
        <layout class="QVBoxLayout" name="verticalLayout_5">
         <item>
          <widget class="QWidget" name="widget" native="true">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <layout class="QGridLayout" name="gridLayout_2">
            <item row="8" column="0">
             <widget class="QLabel" name="lblTime">
              <property name="text">
               <string>Time elapsed:</string>
              </property>
             </widget>
            </item>
            <item row="3" column="0">
             <widget class="QSlider" name="hsdKernelSizen">
              <property name="minimum">
               <number>3</number>
              </property>
              <property name="maximum">
               <number>49</number>
              </property>
              <property name="singleStep">
               <number>2</number>
              </property>
              <property name="value">
               <number>25</number>
              </property>
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="tickPosition">
               <enum>QSlider::NoTicks</enum>
              </property>
             </widget>
            </item>
            <item row="4" column="0">
             <widget class="QLabel" name="lblKernelSizen">
              <property name="text">
               <string>Kernel Size n</string>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QLabel" name="lblKernelSizem">
              <property name="text">
               <string>Kernel Size m</string>
              </property>
             </widget>
            </item>
            <item row="7" column="0">
             <widget class="QPushButton" name="btnExecute">
              <property name="text">
               <string>Execute</string>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QSlider" name="hsdKernelSizem">
              <property name="minimum">
               <number>3</number>
              </property>
              <property name="maximum">
               <number>49</number>
              </property>
              <property name="singleStep">
               <number>2</number>
              </property>
              <property name="value">
               <number>25</number>
              </property>
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
             </widget>
            </item>
            <item row="6" column="0">
             <widget class="QWidget" name="widget_4" native="true">
              <layout class="QGridLayout" name="gridLayout">
               <item row="1" column="1">
                <widget class="QLabel" name="lblGaussianSigma">
                 <property name="text">
                  <string>Gaussian Std</string>
                 </property>
                </widget>
               </item>
               <item row="2" column="1">
                <widget class="QDoubleSpinBox" name="dsbGaussianSigma">
                 <property name="decimals">
                  <number>1</number>
                 </property>
                 <property name="minimum">
                  <double>1.000000000000000</double>
                 </property>
                 <property name="maximum">
                  <double>10.000000000000000</double>
                 </property>
                </widget>
               </item>
               <item row="2" column="0">
                <widget class="QDoubleSpinBox" name="dsbGaussianK">
                 <property name="decimals">
                  <number>1</number>
                 </property>
                 <property name="minimum">
                  <double>0.500000000000000</double>
                 </property>
                 <property name="maximum">
                  <double>10.000000000000000</double>
                 </property>
                 <property name="singleStep">
                  <double>0.500000000000000</double>
                 </property>
                </widget>
               </item>
               <item row="1" column="0">
                <widget class="QLabel" name="lblGaussianK">
                 <property name="text">
                  <string>Gaussian Scale</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="0" colspan="2">
                <widget class="QCheckBox" name="ckbGaussainNormalize">
                 <property name="text">
                  <string>Normalization</string>
                 </property>
                 <property name="checked">
                  <bool>true</bool>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </item>
            <item row="0" column="0">
             <widget class="QComboBox" name="cbbFilter">
              <property name="currentIndex">
               <number>0</number>
              </property>
              <item>
               <property name="text">
                <string>Gaussian Filter</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Box Filter</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Sobel Horizontal FIlter</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Sobel Vertical Filter</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Laplacian Half FIlter</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Laplacian Full FIlter</string>
               </property>
              </item>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <spacer name="verticalSpacer_4">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeType">
            <enum>QSizePolicy::Fixed</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>200</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="tab_2">
        <attribute name="title">
         <string>Edge Detection</string>
        </attribute>
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <item>
          <widget class="QWidget" name="widget_5" native="true">
           <layout class="QGridLayout" name="gridLayout_3">
            <item row="1" column="0">
             <widget class="QLabel" name="lblGaussianSigma_2">
              <property name="text">
               <string>Gaussian Std</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QLabel" name="lblZeroCrossingThreshold">
              <property name="text">
               <string>Zero-crossing Threshold</string>
              </property>
             </widget>
            </item>
            <item row="4" column="0">
             <widget class="QLabel" name="lblKernelSizem_2">
              <property name="text">
               <string>Kernel Size m</string>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QDoubleSpinBox" name="dsbGaussianSigma_2">
              <property name="decimals">
               <number>1</number>
              </property>
              <property name="minimum">
               <double>1.000000000000000</double>
              </property>
              <property name="maximum">
               <double>10.000000000000000</double>
              </property>
             </widget>
            </item>
            <item row="8" column="0">
             <widget class="QLabel" name="lblTime_3">
              <property name="text">
               <string>Time elapsed:</string>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <widget class="QDoubleSpinBox" name="dsbZeroCrossingThreshold">
              <property name="decimals">
               <number>1</number>
              </property>
              <property name="minimum">
               <double>0.000000000000000</double>
              </property>
              <property name="maximum">
               <double>255.000000000000000</double>
              </property>
              <property name="value">
               <double>0.000000000000000</double>
              </property>
             </widget>
            </item>
            <item row="0" column="0" colspan="2">
             <widget class="QLabel" name="lblMH">
              <property name="text">
               <string>The Marr-Hildreth edge detection method</string>
              </property>
             </widget>
            </item>
            <item row="3" column="0" colspan="2">
             <widget class="QSlider" name="hsdKernelSizem_2">
              <property name="minimum">
               <number>3</number>
              </property>
              <property name="maximum">
               <number>49</number>
              </property>
              <property name="singleStep">
               <number>2</number>
              </property>
              <property name="pageStep">
               <number>2</number>
              </property>
              <property name="value">
               <number>25</number>
              </property>
              <property name="tracking">
               <bool>true</bool>
              </property>
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="invertedAppearance">
               <bool>false</bool>
              </property>
              <property name="invertedControls">
               <bool>false</bool>
              </property>
              <property name="tickPosition">
               <enum>QSlider::NoTicks</enum>
              </property>
             </widget>
            </item>
            <item row="5" column="0" colspan="2">
             <widget class="QSlider" name="hsdKernelSizen_2">
              <property name="minimum">
               <number>3</number>
              </property>
              <property name="maximum">
               <number>49</number>
              </property>
              <property name="singleStep">
               <number>2</number>
              </property>
              <property name="value">
               <number>25</number>
              </property>
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
             </widget>
            </item>
            <item row="6" column="0">
             <widget class="QLabel" name="lblKernelSizen_2">
              <property name="text">
               <string>Kernel Size n</string>
              </property>
             </widget>
            </item>
            <item row="7" column="0">
             <widget class="QPushButton" name="btnLoG">
              <property name="text">
               <string>Execute</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QWidget" name="widget_8" native="true">
           <layout class="QGridLayout" name="gridLayout_4">
            <item row="1" column="0">
             <widget class="QPushButton" name="btnSobel">
              <property name="text">
               <string>Execute</string>
              </property>
             </widget>
            </item>
            <item row="0" column="0">
             <widget class="QLabel" name="lblSobel">
              <property name="text">
               <string>Sobel vertical+horizontal</string>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QLabel" name="lblTime_4">
              <property name="text">
               <string>Time elapsed:</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <spacer name="verticalSpacer_2">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeType">
            <enum>QSizePolicy::Fixed</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>218</width>
             <height>150</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="tab_3">
        <attribute name="title">
         <string>Order-statistics</string>
        </attribute>
        <layout class="QVBoxLayout" name="verticalLayout_4">
         <item>
          <widget class="QWidget" name="widget_6" native="true">
           <layout class="QGridLayout" name="gridLayout_6">
            <item row="0" column="0">
             <widget class="QSlider" name="hsdKernelSizem_3">
              <property name="minimum">
               <number>3</number>
              </property>
              <property name="maximum">
               <number>11</number>
              </property>
              <property name="singleStep">
               <number>2</number>
              </property>
              <property name="value">
               <number>7</number>
              </property>
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="lblKernelSizem_3">
              <property name="text">
               <string>Kernel Size m</string>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QSlider" name="hsdKernelSizen_3">
              <property name="minimum">
               <number>3</number>
              </property>
              <property name="maximum">
               <number>11</number>
              </property>
              <property name="singleStep">
               <number>2</number>
              </property>
              <property name="value">
               <number>7</number>
              </property>
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
             </widget>
            </item>
            <item row="3" column="0">
             <widget class="QLabel" name="lblKernelSizen_3">
              <property name="text">
               <string>Kernel Size n</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QWidget" name="widget_7" native="true">
           <layout class="QGridLayout" name="gridLayout_5">
            <item row="0" column="0">
             <widget class="QLabel" name="label">
              <property name="text">
               <string>Order-statistic filter</string>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QRadioButton" name="rbnMedian">
              <property name="text">
               <string>Median</string>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QRadioButton" name="rbnMax">
              <property name="text">
               <string>Max</string>
              </property>
             </widget>
            </item>
            <item row="3" column="0">
             <widget class="QRadioButton" name="rbnMin">
              <property name="text">
               <string>Min</string>
              </property>
             </widget>
            </item>
            <item row="4" column="0">
             <widget class="QPushButton" name="btnOrderStatisticsExecute">
              <property name="text">
               <string>Execute</string>
              </property>
             </widget>
            </item>
            <item row="5" column="0">
             <widget class="QLabel" name="lblTime_2">
              <property name="text">
               <string>Time elapsed:</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <spacer name="verticalSpacer_3">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeType">
            <enum>QSizePolicy::Fixed</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>218</width>
             <height>250</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </widget>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1062</width>
     <height>24</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
